src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -222,3 +353,22 @@ def visit(self, visit_node: Callable[[DependencyGraph], bool | None]) -> bool |
             if dependency.visit(visit_node):
                 return True
         return False
+
+    def build_python_source_dependency_graph(self, python_code: str, register_dependency: Callable[[str], typing.Any]):
```

Comment:
```suggestion
    def build_graph_from_source(self, python_code: str, register_dependency: Callable[[str], typing.Any]):
```
nit: or something shorter. we already have `Dependency` in the class name